src/com/palmergames/bukkit/towny/object/Resident.java

```diff
@@ -378,6 +379,7 @@ public void setTown(Town town) throws AlreadyRegisteredException {
 		this.town = town;
 		updatePerms();
 		town.addResident(this);
+		setJoinedTownAt(System.currentTimeMillis());
```

Comment:
This method is called during loading so all residents would end up having their join dates set to the server load up.